### PR TITLE
Dependent Field optimizations

### PR DESCRIPTION
**Pull request by Hrishabh17**

### CODE REVIEW

File: apps/sage300/dependent_fields.py [hunk 1]
```diff
@@ -96,14 +96,18 @@ def post_dependent_cost_code(dependent_field_setting: DependentFieldSetting, pla
...
+            except Exception as exception:
+                logger.error(f'Exception while posting dependent cost code | Error: {exception} | Payload: {payload}')
+                raise
 
     return posted_cost_codes
 
 
 def post_dependent_cost_type(dependent_field_setting: DependentFieldSetting, platform: PlatformConnector, filters: Dict):
-    cost_categories = CostCategory.objects.filter(**filters).values('cost_code_name').annotate(cost_categories=ArrayAgg('name', distinct=True))
+    cost_categories = CostCategory.objects.filter(is_imported=False, **filters).values('cost_code_name').annotate(cost_categories=ArrayAgg('name', distinct=True))
```
Comment (ashwin1111) [inline apps/sage300/dependent_fields.py:110]: after prod deploy we should run script and set right values to is_imported col

File: scripts/003_update_is_imported_flag.sql [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+inner join dependent_field_settings dfs 
+on cs.workspace_id = dfs.workspace_id
+where cs.updated_at >= dfs.last_successful_import_at 
+and cs.is_imported = 'f';
+
+update cost_category cs
+set is_imported = 't'
+from dependent_field_settings dfs
+where cs.workspace_id = dfs.workspace_id
+and cs.updated_at >= dfs.last_successful_import_at
```
Comment (ashwin1111) [inline scripts/003_update_is_imported_flag.sql:14]: shouldn't this be <= ?

Comment (Hrishabh17) [reply scripts/003_update_is_imported_flag.sql]: fixed it

File: apps/sage300/dependent_fields.py (hunk 1 above)
Comment (Hrishabh17) [reply apps/sage300/dependent_fields.py:110]: have added the script

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**